### PR TITLE
Updated the deploy for AWS

### PR DESCRIPTION
**Pull request by prajwaliam**
# SUMMARY

This PR consists of multiple jira issues, 

PRIM-3197 : Remove all secrets from YAML files for AWS (https://nulli2020.atlassian.net/browse/PRIM-3197). This is done to remove all the hardcoded values in configmap and secrets into terraform values

PRIM-3217 : Remove hardcoding for NiamClient secret everywhere (https://nulli2020.atlassian.net/browse/PRIM-3217) This is done to improve the security by removing the default values for niamclient 

PRIM-3198 : Fix Terraform plan (https://nulli2020.atlassian.net/browse/PRIM-3198). Terraform plan was not functioning accurately, due to cert-manager dependancy issue. So, made sure that cert-manger creates challenge only after the EKS cluster is created

# TEST PLAN
PRIM-3197 : Check the variables.tf file. If it contains a sensitive hardcoded value or not. 

PRIM-3217 : Check if we are using the default values for shared_secret_OIDC_CLIENT_PWD, shared_secret_INTERNAL_SECRET 

PRIM-3198 : Check for terrafrom plan running twice in the ./run.sh

### CODE REVIEW

File: terraform/aws/modules/cypher/main.tf [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
... ON CREATE SET client2._id = uuids[1] MERGE (client3:NiamClient {name: \"3edgesDataproxyClient\"}) ON CREATE SET client3._id = uuids[2] MERGE (client4:NiamClient {name: \"3edgesDataloaderClient\"}) ON CREATE SET client4._id = uuids[3] MERGE (client5:NiamClient {name: \"3edgesDataproxyAuthzClient\"}) ON CREATE SET client5._id = uuids[4] WITH client1, client2, client3, client4, client5 MATCH (n:NiamClient) SET n.grant_types = [\"password\",\"implicit\",\"device_code\",\"social\",\"authorization_code\",\"client_credentials\",\"refresh_token\"], n.redirect_uris = [\"http://localhost:3005\", \"http://localhost:3005/code\",\"https://${var.hosted_zone}\", \"https://${var.hosted_zone}/code\", \"https://webloader.${var.hosted_zone}/code\"], n.post_logout_redirect_uris = [\"http://localhost:3005\",\"http://localhost:3005/login\", \"https://${var.hosted_zone}\", \"https://webloader.${var.hosted_zone}\", \"https://${var.hosted_zone}/login\", \"https://webloader.${var.hosted_zone}/login\"], n.response_types = [\"id_token\",\"code\",\"code id_token\"], n.token_endpoint_auth_method = \"client_secret_basic\", n.client_secret = \"U2FsdGVkX1+BhmwNe9+bzpTlx6/JHyghIyhxeDYPlcQ=\", n.client_id = n.name"
```
Comment (aboteganulli) [inline terraform/aws/modules/cypher/main.tf]: I'm seeing a password here: "n.client_secret" 
You must use this value direct from your variables, right?

File: terraform/aws/modules/kubernetes/deployments/ingress.tf [hunk 2]
```diff
@@ -12,12 +12,12 @@ resource "kubernetes_ingress_v1" "three_edges_ingress" {
...
 
     tls {
-      hosts       = ["*.${var.hosted_zone}"]
+      hosts       = [var.hosted_zone, "*.${var.hosted_zone}"]
       secret_name = "letsencrypt-wildcard-secret"
     }
 
     rule {
-      host = var.hosted_zone
+      host = "frontend.${var.hosted_zone}"
```
Comment (aboteganulli) [inline terraform/aws/modules/kubernetes/deployments/ingress.tf]: This "frontend" was a test, you can remove it from: Pod, Service and Ingress

File: terraform/aws/modules/cypher/main.tf (hunk 1 above)
Comment (prajwaliam) [reply terraform/aws/modules/cypher/main.tf]: I have made changes

File: terraform/aws/modules/kubernetes/deployments/ingress.tf (hunk 2 above)
Comment (prajwaliam) [reply terraform/aws/modules/kubernetes/deployments/ingress.tf]: I have removed the frontend pod, svc, ingress rule

**Review by aboteganulli — APPROVED** [review]

**Pull request merged**